src/main/java/eu/yals/ui/HomeView.java

```diff
@@ -0,0 +1,443 @@
+  }
+
+  private int calculateQRCodeSize() {
+    int[] browserWidthInfo = new int[1];
+    if (getUI().isPresent()) {
+      int[] finalBrowserWidthInfo = browserWidthInfo;
+      getUI()
+          .get()
+          .getPage()
+          .retrieveExtendedClientDetails(
+              details -> finalBrowserWidthInfo[0] = details.getScreenWidth());
+    } else {
+      browserWidthInfo = new int[] {0};
+    }
+    int browserWidth = browserWidthInfo[0];
+
+    int defaultQRBlockSize = 371;
+    int defaultQRCodeSize = 350;
+    float qrBlockRatio = 0.943f; // 350/371
+
+    int size;
+    if (browserWidth > defaultQRBlockSize) {
+      size = defaultQRCodeSize;
+    } else {
+      size = Math.round(browserWidth * qrBlockRatio);
+    }
+    return size;
+  }
+
+  private void generateQRCode(String ident) {
+    int size = calculateQRCodeSize();
+    String qrCodeGeneratorRoute;
+    if (size == 0) {
+      qrCodeGeneratorRoute =
+          String.format("%s/%s/%s", appUtils.getAPIHostPort(), Endpoint.Api.QR_CODE_API, ident);
+    } else {
+      qrCodeGeneratorRoute =
+          String.format(
+              "%s/%s/%s/%d", appUtils.getAPIHostPort(), Endpoint.Api.QR_CODE_API, ident, size);
+    }
+
+    HttpResponse<JsonNode> response = Unirest.get(qrCodeGeneratorRoute).asJson();
+    if (response.isSuccess()) {
+      onSuccessGenerateQRCode(response);
+    } else {
+      onFailGenerateQRCode(response);
+    }
+  }
+
+  private void onSuccessGenerateQRCode(HttpResponse<JsonNode> response) {
+    if (response.getStatus() == 200) {
+      String qrCode = response.getBody().getObject().getString("qrCode");
+      if (StringUtils.isNotBlank(qrCode)) {
+        this.qrCode.setSrc(qrCode);
+        qrCodeRow.setVisible(true);
+      } else {
+        showError("Internal error. Got malformed reply from QR generator");
+      }
+    }
+  }
+
+  private void onFailGenerateQRCode(HttpResponse<JsonNode> response) {
+    showError("Internal error. Got malformed reply from QR generator");
+    this.qrCode.setSrc("");
+    qrCodeRow.setVisible(false);
+
+    if (response.getBody() != null) {
+      log.debug("{} QR Code Reply JSON: {}", TAG, response.getBody());
+    }
+  }
+
+  private void showError(String errorMessage) {
+    errorNotification = getErrorNotification(errorMessage);
+    errorNotification.open();
+  }
+
+  private void cleanForm() {
+    input.setValue("");
+    updateButtonState();
+  }
+
+  private void cleanErrors() {
+    if (errorNotification != null && errorNotification.isOpened()) {
+      errorNotification.close();
+    }
+  }
+
+  private void cleanResults() {
+    shortLink.setHref("");
+    shortLink.setText("");
+    resultRow.setVisible(false);
+
+    qrCode.setSrc("");
+    qrCodeRow.setVisible(false);
+  }
+
+  private void updateButtonState() {
+    boolean inputHasValue = StringUtils.isNotBlank(input.getValue());
+    submitButton.setEnabled(inputHasValue);
+  }
+
+  public static class IDs {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The utility class name 'IDs' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)